Engine/DataFeeds/SubscriptionSynchronizer.cs

```diff
@@ -108,40 +106,47 @@ public TimeSlice Sync(DateTime frontier, IEnumerable<Subscription> subscriptions
-                            }
-                            else if (packetBaseDataCollection is FuturesChainUniverseDataCollection)
+                            // assume that if the first item is a base data collection then the enumerator handled the aggregation,
+                            // otherwise, load all the the data into a new collection instance
+                            var packetBaseDataCollection = packet.Data[0] as BaseDataCollection;
+                            var packetData = packetBaseDataCollection == null
+                                ? packet.Data
+                                : packetBaseDataCollection.Data;
+
+                            BaseDataCollection collection;
+                            if (!universeData.TryGetValue(subscription.Universe, out collection))
                             {
-                                var current = subscription.Current as FuturesChainUniverseDataCollection;
-                                collection = new FuturesChainUniverseDataCollection(frontier, subscription.Configuration.Symbol, packetData);
+                                if (packetBaseDataCollection is OptionChainUniverseDataCollection)
+                                {
+                                    var current = subscription.Current as OptionChainUniverseDataCollection;
+                                    var underlying = current != null ? current.Underlying : null;
+                                    collection = new OptionChainUniverseDataCollection(frontier, subscription.Configuration.Symbol, packetData, underlying);
+                                }
+                                else if (packetBaseDataCollection is FuturesChainUniverseDataCollection)
+                                {
+                                    var current = subscription.Current as FuturesChainUniverseDataCollection;
+                                    collection = new FuturesChainUniverseDataCollection(frontier, subscription.Configuration.Symbol, packetData);
+                                }
+                                else
+                                {
+                                    collection = new BaseDataCollection(frontier, subscription.Configuration.Symbol, packetData);
+                                }
+
+                                universeData[subscription.Universe] = collection;
                             }
                             else
                             {
-                                collection = new BaseDataCollection(frontier, subscription.Configuration.Symbol, packetData);
+                                collection.Data.AddRange(packetData);
                             }
-
-                            universeData[subscription.Universe] = collection;
                         }
                         else
                         {
-                            collection.Data.AddRange(packetData);
+                            data.Add(packet);
```

Comment:
no changed required, but personal style, if an else block has a single line I typically invert the if so that the single line is up top near the condition and the larger block is in the else. This helps keeps the condition closer to it's downstream impact